### PR TITLE
Use type application for 'decode'

### PR DESCRIPTION
**Pull request by shayne-fletcher**
TsCodeGen - replace a type annotation by use of a type application.

### CODE REVIEW

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 1]
```diff
@@ -743,7 +743,7 @@ setupWorkspace optInputPackageJson optOutputDir dependencies = do
     (Just <$> BSL.readFile optInputPackageJson) (const $ pure Nothing)
   packageJson <- case mbBytes of
     Nothing -> pure mempty
-    Just bytes -> case decode bytes :: Maybe PackageJson of
+    Just bytes -> case decode @PackageJson bytes of
```
Comment (cocreature) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: If you use `eiherDecode` you can get an error message on parse failures.

Comment (hurryabit) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Excellent idea. Would be `eitherDecode @PackageJson` then I guess.

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: I'll get another PR going for that.

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: I've had a look at the error messages it produces. They don't look helpful to be honest. For example, with a stray ',' the resulting error message is:
```
Error in $: Failed reading: satisfy. Expecting object value
```
I recommend we do not switch from `decode` to `eitherDecode`.

Comment (cocreature) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: It depends on the error and the kind of code you write. E.g. for a missing key you can often get fairly decent errors. Using things like `withObject`, `<?>` you can improve errors. I wouldn’t use one bad error message as an indicator that not showing the error at all is better.

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Ok, sounds reasonable. In which case I offer https://github.com/digital-asset/daml/pull/5098 😃

**Pull request merged**